SECURITY_RELEASE_PROCESS.md

```diff
@@ -0,0 +1,251 @@
+  the fix to users environments; this can include links to external distributor documentation.
+- The Fix Lead will remove the Fix Team from the private security repo.
+
+### Retrospective
+
+These steps should be completed 1-3 days after the Release Date. The retrospective process
+[should be blameless](https://landing.google.com/sre/book/chapters/postmortem-culture.html).
+
+- The Fix Lead will send a retrospective of the process to envoy-dev@googlegroups.com including
+  details on everyone involved, the timeline of the process, links to relevant PRs that introduced
+  the issue, if relevant, and any critiques of the response and release process.
+- Maintainers and Fix Team are also encouraged to send their own feedback on the process to
+  envoy-dev@googlegroups.com. Honest critique is the only way we are going to get good at this as a
+  community.
+
+## Private Distributors List
+
+This list is intended to be used primarily to provide actionable information to
+multiple distribution vendors at once. This list is not intended for
+individuals to find out about security issues.
+
+### Embargo Policy
+
+The information members receive on envoy-distributors-announce must not be made public, shared, nor
+even hinted at anywhere beyond the need-to-know within your specific team except with the list's
+explicit approval. This holds true until the public disclosure date/time that was agreed upon by the
+list. Members of the list and others may not use the information for anything other than getting the
+issue fixed for your respective distribution's users.
+
+Before any information from the list is shared with respective members of your team required to fix
+said issue, they must agree to the same terms and only find out information on a need-to-know basis.
+
+In the unfortunate event you share the information beyond what is allowed by this policy, you _must_
+urgently inform the envoy-security@googlegroups.com mailing list of exactly what information leaked
+and to whom. A retrospective will take place after the leak so we can assess how to not make the
+same mistake in the future.
+
+If you continue to leak information and break the policy outlined here, you will be removed from the
+list.
+
+### Contributing Back
+
+This is a team effort. As a member of the list you must carry some water. This
+could be in the form of the following:
+
+**Technical**
+
+- Review and/or test the proposed patches and point out potential issues with
+  them (such as incomplete fixes for the originally reported issues, additional
+  issues you might notice, and newly introduced bugs), and inform the list of the
+  work done even if no issues were encountered.
+
+**Administrative**
+
+- Help draft emails to the public disclosure mailing list.
+- Help with release notes.
+
+### Membership Criteria
+
+To be eligible for the envoy-distributors-announce mailing list, your
+distribution should:
```

Comment:
I think this list could possibly use some clarification.  I absolutely agree we want to make sure random folks don't sign up for envoy-distributors-announce since if they don't have skin in the game there's a higher chance of leaks.  

To walk through, I'm not sure what an actively maintained distribution means.  Have active Envoy deployments?  
For user-base, don't we care about organizations using Envoy internally who are convened about insider threats? 
Is there a chicken-and-egg problem with fixing security issues or you basically apply them once they're public and eventually get the advanced notice?
I think for now, being an active participant in the community and having Envoy deployments (plus 6/7) are key.

Reply:
> To walk through, I'm not sure what an actively maintained distribution means. Have active Envoy deployments?

Historically (and I am not an expert on this), distributor lists are typically only joinable by entities that package and redistribute the code. For example, Ubuntu, Red Hat, etc. It gets hazy if for example Amazon asks to join on behalf of Amazon Linux. Technically they should be allowed to join. Personally, I would say that we should allow typical distributors to join as well as vendors that are *publicly* running Envoy on behalf of customers in a cloud environment. This in my opinion is an extra incentive to step up and be a maintainer: all maintainers will have early access to security issues. Anyhow, thanks for bringing this up and this is something we are going to need to iterate on a bit. Curious to hear people's thoughts on this and I will send an email on this also. cc @caniszczyk. 

> For user-base, don't we care about organizations using Envoy internally who are convened about insider threats?

We care, but FWIW this is not typically criteria sufficient to get on this type of list.

> Is there a chicken-and-egg problem with fixing security issues or you basically apply them once they're public and eventually get the advanced notice?

Can you clarify? Are you asking how someone would know what patch to apply without it being public? Typically anyone on this list would either be a packager, and ready to release a new version on disclosure day, or a large vendor, how would pre-patch their services before disclosure day.

Reply:
Just sanity checking that the way to get access to security fixes early is start out applying them the day they're released, then once you build a track record you can get the early access.

Reply:
> Just sanity checking that the way to get access to security fixes early is start out applying them the day they're released, then once you build a track record you can get the early access.

I think for orgs that already have a track record of working within this type of process (e.g., again Ubuntu and Red Hat) they can just get immediately on this list most likely. But yes, in general, a track record of being a good citizen is important.